integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/UsedImagesTest.java

```diff
@@ -0,0 +1,171 @@
+        String freeIpaName = resourcePropertyProvider().getName();
+        String sdxName = resourcePropertyProvider().getName();
+        CloudbreakUser internalActor = cloudbreakActor.createInternal(testContext.getActingUserCrn().getAccountId());
+
+        testContext
+                .given(ImageCatalogTestDto.class)
+                    .withName(cbImageCatalogName)
+                    .withUrl(getImageCatalogMockServerSetup().getPreWarmedImageCatalogUrl())
+                .when(imageCatalogTestClient.createV4(), key(cbImageCatalogName))
+                .given(EnvironmentNetworkTestDto.class)
+                .given(envName, EnvironmentTestDto.class)
+                    .withNetwork()
+                    .withCreateFreeIpa(Boolean.FALSE)
+                .when(environmentTestClient.create(), key(envName))
+                .await(EnvironmentStatus.AVAILABLE, key(envName))
+                .given(freeIpaName, FreeIpaTestDto.class)
+                    .withCatalog(getImageCatalogMockServerSetup().getFreeIpaImageCatalogUrl())
+                .when(freeIpaTestClient.create(), key(freeIpaName))
+                .await(AVAILABLE, key(freeIpaName))
+                .given(sdxName, SdxInternalTestDto.class)
+                    .withImageCatalogNameAndImageId(cbImageCatalogName, SDX_IMAGE_UUID)
+                .when(sdxTestClient.createInternal(), key(sdxName))
+                .await(SdxClusterStatusResponse.RUNNING, key(sdxName))
+                .then((tc, testDto, client) -> verifyImagesAreUsed(tc, testDto, internalActor))
+                .validate();
+        testContext
+                .as(cloudbreakActor.defaultUser())
+                .given(envName, EnvironmentTestDto.class)
+                .when(environmentTestClient.delete(), key(envName))
+                .await(EnvironmentStatus.ARCHIVED, key(envName))
+                .then((tc, testDto, client) -> verifySdxImageIsNotUsed(tc, testDto, internalActor))
+                .validate();
+    }
+
+    private <T extends CloudbreakTestDto> T verifyImagesAreUsed(TestContext testContext, T testDto, CloudbreakUser internalActor) {
+        testContext
+                .as(internalActor)
+                .given(UsedImagesTestDto.class)
+                .when(utilTestClient.usedImages())
+                .then((tc, usedImagesTestDto, client) -> {
+                    List<UsedImageStacksV4Response> usedImages = usedImagesTestDto.getResponse().getUsedImages();
+                    UsedImageStacksV4Response usedImageStacksV4Response = usedImages.stream()
+                            .filter(usedImage -> usedImage.getImage().getImageId().contains(SDX_IMAGE_UUID))
+                            .findFirst().orElseThrow(() -> new TestFailException(String.format("SDX image is NOT in use with ID:: %s", SDX_IMAGE_UUID)));
+                    LOGGER.info("Used SDX image ID:: {}", usedImageStacksV4Response.getImage().getImageId());
```

Comment:
TestFailException is needed, because of `Assert` just generate an error, these errors won't appear in the test results. So test can show as passed even the related assertions are failing. However test context exception map is gathering all the exception during test and finally these are shown in the test log and test results as well.

Reply:
I think this should be fixed in the test framework, but this will be fine for now.